ldms/src/ldmsd/ldmsd_config.c

```diff
@@ -309,20 +327,56 @@ void ldmsd_store___del(ldmsd_cfgobj_t obj)
 int ldmsd_load_plugin(char* cfg_name, char *plugin_name,
 		      char *errstr, size_t errlen)
 {
+	const char *libpath;
 	struct ldmsd_plugin *api;
+	ldmsd_cfgobj_sampler_t sampler;
+	ldmsd_cfgobj_store_t store;
+	char log_name[1024];
+	ldmsd_cfgobj_t cfgobj;
+
 	if (!plugin_name || !cfg_name)
 		return EINVAL;
-	api = load_plugin(plugin_name);
+	/* If this plugin does not support multiple configurations,
+	 * the plugin_name must equal the cfg_name
+	 */
+	api = load_plugin(plugin_name, &libpath);
 	if (!api)
 		return errno;
+	if (0 == (api->flags & LDMSD_PLUGIN_MULTI_INSTANCE)) {
+		if (cfg_name && plugin_name) {
```

Comment:
This test is redundant. This function enforces that they are both set a few lines earlier.